test/workflows/test_workflow.py

```diff
@@ -409,26 +266,159 @@ def test_output_details_with_long(self, mock_click, mock_prose_print):
         )
         mock_prose_print.called_once_with("description", CONSOLE_WIDTH)
 
+    @patch("dafni_cli.workflows.workflow.format_table")
+    def test_format_parameter_sets(
+        self,
+        mock_format_table,
+    ):
+        """Tests format_parameter_sets works correctly"""
+        # SETUP
+        workflow: Workflow = parse_workflow(TEST_WORKFLOW)
+
+        # Two identical parameter sets
+        workflow.parameter_sets.append(workflow.parameter_sets[0])
+
+        # CALL
+        result = workflow.format_parameter_sets()
+
+        # ASSERT
+        mock_format_table.assert_called_once_with(
+            headers=[
+                TABLE_ID_HEADER,
+                TABLE_NAME_HEADER,
+                TABLE_PUBLISHED_BY_HEADER,
+                TABLE_PUBLISHED_DATE_HEADER,
+            ],
+            rows=[
+                [
+                    "0a0a0a0a-0a00-0a00-a000-0a0a0000000a",
+                    "First parameter set",
+                    "Joel Davies",
+                    "2023-04-04",
+                ],
+            ]
+            * 2,
+        )
+        self.assertEqual(result, mock_format_table.return_value)
+
+    @patch("dafni_cli.workflows.workflow.format_table")
+    def test_format_instances(
+        self,
+        mock_format_table,
+    ):
+        """Tests format_instances works correctly"""
+        # SETUP
+        workflow: Workflow = parse_workflow(TEST_WORKFLOW)
+
+        # Two identical instances
+        workflow.instances.append(workflow.instances[0])
+
+        # CALL
+        result = workflow.format_instances()
+
+        # ASSERT
+        mock_format_table.assert_called_once_with(
+            headers=[
+                TABLE_ID_HEADER,
+                TABLE_WORKFLOW_VERSION_ID_HEADER,
+                TABLE_PARAMETER_SET_HEADER,
+                TABLE_STARTED_HEADER,
+                TABLE_FINISHED_HEADER,
+                TABLE_STATUS_HEADER,
+            ],
+            rows=[
+                [
+                    "0a0a0a0a-0a00-0a00-a000-0a0a0000000c",
+                    "0a0a0a0a-0a00-0a00-a000-0a0a0000000b",
+                    "First parameter set",
+                    "2023-04-06T12:46:38",
+                    "2023-04-06T12:58:35",
+                    "Succeeded",
+                ],
+            ]
+            * 2,
+        )
+        self.assertEqual(result, mock_format_table.return_value)
+
+    @patch("dafni_cli.workflows.workflow.prose_print")
+    @patch("dafni_cli.workflows.workflow.click")
+    @patch.object(Workflow, "format_parameter_sets")
+    @patch.object(Workflow, "format_instances")
+    def test_output_info(
+        self,
+        mock_format_instances,
+        mock_format_parameter_sets,
+        mock_click,
+        mock_prose_print,
+    ):
+        """Tests output_info works correctly"""
+        # SETUP
+        workflow = parse_workflow(TEST_WORKFLOW)
+
+        workflow.output_info()
```

Comment:
Missing your comments here separating up the call and assert sections of the test